lib/galaxy/tools/deps/mulled/util.py

```diff
@@ -0,0 +1,108 @@
+"""Utilities for working with mulled abstractions outside the mulled package."""
+from __future__ import print_function
+
+import collections
+import hashlib
+
+from distutils.version import LooseVersion
+
+try:
+    import requests
+except ImportError:
+    requests = None
+
+
+def quay_versions(namespace, pkg_name):
+    """Get all version tags for a Docker image stored on quay.io for supplied package name."""
+    if requests is None:
+        raise Exception("requets library is unavailable, functionality not available.")
+
+    url = 'https://quay.io/api/v1/repository/%s/%s' % (namespace, pkg_name)
+    response = requests.get(url, timeout=None)
```

Comment:
fwiw, I'm using something like https://gist.github.com/mvdbeek/5e48d2c21c1440e22bda895769f5e450 for v2 registries.